cogs/wiki.py

```diff
@@ -38,42 +44,90 @@ def __init__(self, bot: Bot):
         urls = [response['query']['pages'][str(pageid)]['fullurl'] for pageid in pageids]
         return urls
 
+    async def read_titles(self, namespace):
+        """Helper function to read all the page titles from a single namespace.
+        Sandbox link for this query:
+        https://cavesofqud.gamepedia.com/Special:ApiSandbox#action=query&format=json&prop=&list=allpages&pageids=1&apnamespace=0
+        """
+        fresh_titles = {}
+        # there's a limit on how many titles we can fetch at a time (currently 5000 for bots)
+        # and we may have more wiki articles than that someday, so fetch in batches
+        got_all = False
+        apfrom = ''  # the article title to 'continue' querying from, if necessary
+        while not got_all:
+            params = {'format': 'json',
+                      'action': 'query',
+                      'list': 'allpages',
+                      'apfrom': apfrom,
+                      'apnamespace': namespace,
+                      'aplimit': 5000}  # TODO: add to config
+            async with http_session.get(url=self.url, params=params) as reply:
+                response = await reply.json()
+            new_items = response['query']['allpages']
+            for item in new_items:
+                title = item['title']
+                # filter out some Cargo tables that shouldn't be in main namespace
+                if not title.startswith('DynamicObjectsTable:'):
```

Comment:
Everything is currently moved out of that prefix in order to make room for custom namespaces (eventually). I moved them to TEMP(pagename) so that should be the string to check here.

Reply:
Done, while doing this I found that the reason they were in the list in the first place was that I was including redirects in the title getting query. Fixed this by including `apfilterredir=nonredirects` and deleted the extra redirects on the wiki.